src/cortex/compute/nn/activations.clj

```diff
@@ -0,0 +1,57 @@
+(ns cortex.compute.nn.activations
+  "High level implemenations of activations that work across all backends"
+  (:require [clojure.core.matrix :as m]
+            [cortex.tensor :as tensor]
+            [think.datatype.core :as dtype]))
+
+
+;;;; Forward
+
+(defn logistic [input output]
+  (tensor/unary-op! output 1.0 input :logistic))
+
+(defn tanh [input output]
+  (tensor/unary-op! output 1.0 input :tanh))
+
+(defn relu [input output]
+  (tensor/binary-op! output 1.0 input 0 0 :max))
+
+(defn swish
+  "sigmoid(x)*x"
+  [input output]
+  (tensor/unary-op! output 1.0 input :logistic)
+  (tensor/binary-op! output 1.0 input 1.0 output :*))
+
+;;; Backwards
+
+(defn default-gradient
+  "Provides the default gradient for tanh, logistic, and relu"
+  [input-gradient output-gradient output act-type]
+  (tensor/activation-gradient! input-gradient output-gradient output act-type))
+
+(defn logistic-gradient [input-gradient output-gradient output]
+  (default-gradient input-gradient output-gradient output :logistic))
+
+(defn tanh-gradient [input-gradient output-gradient output]
+  (default-gradient input-gradient output-gradient output :tanh))
+
+(defn relu-gradient [input-gradient output-gradient output]
+  (default-gradient input-gradient output-gradient output :relu))
+
+(defn swish-gradient
+  "(fx + sigm *(1 -fx)) * output-grad - where fx = sigm(x) * x"
+  [input-gradient output-gradient output]
+  (let [fx (tensor/new-tensor (m/shape output) :datatype (dtype/get-datatype output))
```

Comment:
Unsure of whether it's better to have the implementation here in terms of the base unary operators or add `swish` and `swish-grad` as new unary-operators with https://github.com/thinktopic/cortex/blob/master/resources/operations.h (for cuda). Happy to convert it to a new unary operator if you think it works better. (I had trouble running nvcc on my local computer so I tried this way first :) )